File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/jmvp/ui/search/home/SearchPresenter.java
package com.woowrale.jmvp.ui.search.home;

import com.woowrale.jmvp.ui.base.BasePresenter;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class SearchPresenter extends BasePresenter<SearchView> {

    @Inject
    public SearchPresenter() {

    }

    public void navigateToSearch(String searchType){
        if (searchType.equals("L")){
            getView().navigateToSearchLocal();
        }else{
            getView().navigateToSearchRemote();
        }
    }
}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/mvp/ui/search/home/SearchPresenter.kt
package com.woowrale.mvp.ui.search.home

import com.woowrale.mvp.ui.base.BasePresenter
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SearchPresenter @Inject constructor() : BasePresenter<SearchView>() {

    fun navigateToSearch(searchType: String) {
        if (searchType == "L") {
            getView().navigateToSearchLocal()
        } else {
            getView().navigateToSearchRemote()
        }
    }
}
